sample/kotlin/src/main/kotlin/com/datadog/android/sample/datalist/DataListViewModel.kt

```diff
@@ -8,36 +8,56 @@ package com.datadog.android.sample.datalist
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
+import com.datadog.android.rx.publishErrorsToRum
 import com.datadog.android.sample.data.DataRepository
-import com.datadog.android.sample.data.Result
 import com.datadog.android.sample.data.model.Log
-
-class DataListViewModel(val repository: DataRepository) : ViewModel(),
-    DataRepository.Callback<List<Log>> {
-
-    val data = mutableListOf<String>()
-    val liveData = MutableLiveData<UIResponse>()
-
-    fun onAddData() {
-        data.add("Item ${data.size}")
-        repository.getLogs("source:android", this)
+import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers
+import io.reactivex.rxjava3.core.Observable
+import io.reactivex.rxjava3.disposables.Disposable
+import io.reactivex.rxjava3.subjects.PublishSubject
+
+class DataListViewModel(val repository: DataRepository) : ViewModel() {
+
+    private val uiRequestSubject: PublishSubject<UIRequest> = PublishSubject.create()
+    private var disposable: Disposable? = null
+    private val performRequestObservable: Observable<UIResponse> =
+        uiRequestSubject
+            .switchMap { request ->
+                when (request) {
+                    is UIRequest.FetchData -> repository.getLogs("source:android")
+                        .toObservable()
```

Comment:
Is there a real need to convert it from Single to Observable ? If you use `switchMapSingle` instead of `switchMap`, you can avoid this conversion.

Reply:
It is because a Single should only return a value but in our case the `uiRequestObservable` should be an observable because it will deal with UI requests (in this case there's only one but there can be many others which will go in the `when(request)` switch and will transform in a stream.